pkg/frontend/transport/handler_test.go

```diff
@@ -49,35 +49,49 @@ func TestWriteError(t *testing.T) {
 }
 
 func TestHandler_ServeHTTP(t *testing.T) {
+	statusOKRoundTripper := roundTripperFunc(func(req *http.Request) (*http.Response, error) {
+		return &http.Response{
+			StatusCode: http.StatusOK,
+			Body:       io.NopCloser(strings.NewReader("{}")),
+		}, nil
+	})
+
 	for _, tt := range []struct {
-		name            string
-		cfg             HandlerConfig
-		expectedMetrics int
+		name             string
+		cfg              HandlerConfig
+		path             string
+		expectedMetrics  int
+		expectedActivity string
 	}{
 		{
-			name:            "test handler with stats enabled",
-			cfg:             HandlerConfig{QueryStatsEnabled: true},
-			expectedMetrics: 4,
+			name:             "test handler with stats enabled with params",
+			cfg:              HandlerConfig{QueryStatsEnabled: true},
+			path:             "/api/v1/query?query=some_metric&time=42",
+			expectedMetrics:  4,
+			expectedActivity: "12345: GET /api/v1/query params:query=some_metric&time=42",
+		},
+		{
+			name:             "test handler with stats enabled without params",
+			cfg:              HandlerConfig{QueryStatsEnabled: true},
+			path:             "/api/v1/query",
+			expectedMetrics:  4,
+			expectedActivity: "12345: GET /api/v1/query params:(none)",
 		},
 		{
-			name:            "test handler with stats disabled",
-			cfg:             HandlerConfig{QueryStatsEnabled: false},
-			expectedMetrics: 0,
+			name:             "test handler with stats disabled",
+			cfg:              HandlerConfig{QueryStatsEnabled: false},
+			path:             "/api/v1/query?query=some_metric&time=42",
+			expectedMetrics:  0,
+			expectedActivity: "12345: GET /api/v1/query params:query=some_metric&time=42",
 		},
 	} {
 		t.Run(tt.name, func(t *testing.T) {
-			roundTripper := roundTripperFunc(func(req *http.Request) (*http.Response, error) {
-				return &http.Response{
-					StatusCode: http.StatusOK,
-					Body:       io.NopCloser(strings.NewReader("{}")),
-				}, nil
-			})
-
 			reg := prometheus.NewPedanticRegistry()
-			handler := NewHandler(tt.cfg, roundTripper, log.NewNopLogger(), reg)
+			at := testActivityTracker{}
```

Comment:
We could test with real activity tracker -- just do the check from the `statusOKRoundTripper`, and then verify that there are no activities at the end of test. That would simplify the code (no need for mocking interface).

Reply:
> just do the check from the `statusOKRoundTripper`

How can I do that? I haven't seen any way of peeking into the ActivityTracker's status.

Reply:
By using `LoadUnfinishedEntries` function.